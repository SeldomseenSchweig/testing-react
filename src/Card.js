import React from "react";
import "./Card.css";

function Card(props) {
  return (
    <div className="Card">
      <h4 className="Card-title">{props.caption}</h4>
      <img className="Card-image" src={props.src} alt={props.caption} />
      <small className="Card-small">
        Image {props.currNum} of {props.totalNum}.
      </small>
    </div>
  );
}
Card.defaultProps = {
  caption: "hello",
  source: null,
  totalNum:5
};



export default Card;
