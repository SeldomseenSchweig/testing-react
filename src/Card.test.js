import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";


test("smoke test - renders without crashing", ()=>{

    render(<Card />)
  })


test('card should match snapshot', ()=>{

    const { asFragment } = render(<Card />)
    expect(asFragment()).toMatchSnapshot()
  })