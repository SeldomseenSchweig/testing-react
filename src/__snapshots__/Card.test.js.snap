// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      hello
    </h4>
    <img
      alt="hello"
      class="Card-image"
    />
    <small
      class="Card-small"
    >
      Image  of 5.
    </small>
  </div>
</DocumentFragment>
`;
